IoTHubBuddy/IoTViewer/DevicePage.xaml.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+using IoTHubBuddy.Models;
+
+// The Blank Page item template is documented at http://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace IoTHubBuddy
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class DevicePage : Page
+    {
+        private IoTAccountData data;
+        public DevicePage()
+        {
+            this.InitializeComponent();
+        }
+        protected override async void OnNavigatedTo(NavigationEventArgs e)
+        {
+            IoTAccountData account = e.Parameter as IoTAccountData;
+            if (checkValidity(account))
+            {
+                data = account;
+                data.SharedAccessPolicy = "iothubowner";
+                ICollection<string> devices = await IoTDataManager.GetIoTDevices(data.Subscription, data.ResourceGroup, data.HubName, data.SharedAccessPolicy);
+                data.PrimaryKey = await IoTDataManager.GetPrimaryKey(data.Subscription, data.ResourceGroup, data.HubName, data.SharedAccessPolicy);
+                if (devices.Count == 0)
+                {
+                    showError("You have no devices registered in this hub");
+                } else
+                {
+                    hideErrors();
+                    foreach (string dev in devices)
+                    {
+                        DeviceList.Items.Add(dev);
+                    }
+                }
+                
+            } else
+            {
+                showError("An error occurred. Please try logging in again.", true);
+            }
+            
+
+        }
+        private void ItemSelected(object sender, ItemClickEventArgs e)
+        {
+            string device = e.ClickedItem.ToString();
+            data.DeviceName = device;;
+            this.Frame.Navigate(typeof(MapPage), data);
+        }
+        private bool checkValidity(IoTAccountData account)
```

Comment:
nitpick on styling -- showError and hideErrors are in camelCase, other names are PascalCase.